core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -68,6 +68,9 @@ private[spark] class Executor(
 
   // Make sure the local hostname we report matches the cluster scheduler's name for this host
   Utils.setCustomHostname(executorHostname)
+  
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
```

Comment:
unfortunately it doesn't look like `RpcEnv` exposes its port, so we would either need to change that, or use just the `executorId`